File: skills/revelio-python/scripts/load_revelio_extracts.py
"""Small helpers for loading local Revelio extracts in Python."""

from __future__ import annotations

from pathlib import Path

import pandas as pd


def load_table(path: str | Path) -> pd.DataFrame:
    path = Path(path)
    suffix = path.suffix.lower()

    if suffix == ".parquet":
        return pd.read_parquet(path)
    if suffix in {".pkl", ".pickle"}:
        return pd.read_pickle(path)
    if suffix == ".csv":
        return pd.read_csv(path)
    if suffix in {".xlsx", ".xls"}:
        return pd.read_excel(path)

    raise ValueError(f"Unsupported file type: {path.suffix}")


def parse_revelio_dates(df: pd.DataFrame) -> pd.DataFrame:
    out = df.copy()
    for col in ["startdate", "enddate", "updated_dt"]:
        if col in out.columns:
            out[col] = pd.to_datetime(out[col], errors="coerce")
    return out


def assert_required_columns(df: pd.DataFrame, columns: list[str]) -> None:
    missing = sorted(set(columns) - set(df.columns))
    if missing:
        raise ValueError(f"Missing required columns: {missing}")

File: skills/revelio-r/scripts/load_revelio_extracts.R
# Small helpers for loading local Revelio extracts in R.

load_revelio_table <- function(path) {
  ext <- tolower(tools::file_ext(path))

  if (ext == "csv") {
    return(readr::read_csv(path, show_col_types = FALSE))
  }
  if (ext %in% c("xlsx", "xls")) {
    return(readxl::read_excel(path))
  }
  if (ext == "rds") {
    return(readRDS(path))
  }
  if (ext == "parquet") {
    return(arrow::read_parquet(path))
  }

  stop("Unsupported file type: ", ext)
}

parse_revelio_dates <- function(df) {
  date_cols <- intersect(c("startdate", "enddate", "updated_dt"), names(df))
  dplyr::mutate(df, dplyr::across(dplyr::all_of(date_cols), lubridate::ymd))
}

assert_required_columns <- function(df, columns) {
  missing <- setdiff(columns, names(df))
  if (length(missing) > 0) {
    stop("Missing required columns: ", paste(missing, collapse = ", "))
  }
  invisible(TRUE)
}

